simplq/src/components/Layout/Layout.jsx

```diff
@@ -1,24 +1,12 @@
-import React, { useEffect } from 'react';
+import React from 'react';
 import { useAuth0 } from '@auth0/auth0-react';
 import Footer from 'components/common/Footer';
 import Loading from 'components/common/Loading/Loading';
-import { useGetUserQueues } from 'store/asyncActions';
-import { useDispatch } from 'react-redux';
 import Routes from './Routes';
 import styles from './Layout.module.scss';
 
 function Layout() {
-  const { isLoading, isAuthenticated } = useAuth0();
-  const dispatch = useDispatch();
-  const getUserQueues = useGetUserQueues();
-
-  useEffect(() => {
-    // TODO: Server still returns queues for anonymous users
-    //  Dispatch action only if auth is loaded
-    if (isLoading === false) {
-      dispatch(getUserQueues());
-    }
-  }, [isLoading, isAuthenticated, getUserQueues, dispatch]);
+  const { isLoading } = useAuth0();
 
```

Comment:
Since Layout is used in all pages, the active queues were being fetched and the popup was showing up if we directly load any of the other pages which is not needed.

Reply:
Maybe we would want to move this to MyQueues instead of LandingPage, since it's only used in the homePage right now, I left it in LandingPage.

Reply:
The idea here was to move all requests further up in the component tree, so the data is ready no matter what URL the user is coming from. It is a step closer to a single-page app. Currently, the home page will send unnecessary requests to the server on almost every render (as described in #545).

This can be eventually converted into PWA, so everything is loaded only once, and updated as needed.

Since re-render is cheaper than re-fetch, I would put the effort in fixing the behavior (e.g. remove the popup, update store on successful requests).

Reply:
That's a good strategy, but I'm still wondering if that should apply to requests like this one which only make sense in one route and only for admins. We would definitely have more queue members than admins, so do we really want to make the getUserQueues() request trigger for users who just want to join a queue and keep tabs on the status?

Also since this request is only needed in the homepage, do we have to worry about unnecessary re-fetches? On page refresh, it triggers each time, but we would want that for consistency's sake, right?

Reply:
We could add MyQueues to the admin panel, so an admin can quickly switch queues from the admin panel without going to the homepage. And if a user is not the admin, there will be nothing to fetch.

> Also since this request is only needed in the homepage, do we have to worry about unnecessary re-fetches?

Currently, if the user switches between home and admin, the fetch is triggered every time.

Reply:
> Currently, if the user switches between home and admin, the fetch is triggered every time.

Oh yeah, I missed this. Leaving it in the layout is a good idea to prevent this. Good catch. I'll revert this change.

Reply:
I'm thinking of changing the popup behavior by removing the extraReducer in AppSlice and adding a popup trigger inside the MyQueues component. That way, the popup only shows up when needed.

Reply:
> Currently, if the user switches between home and admin, the fetch is triggered every time.

Is this desirable? I think we should fetch every time home page is rendered. So can we dispatch at root of home page?

Reply:
> Is this desirable? I think we should fetch every time home page is rendered.

It would be needed for the admin to see his newly created queue in the list. I was initially thinking of leaving the dispatch in the Layout and manually adding to the queue list for newly created queues, but it's simpler and cleaner to re-fetch on every home page render. It isn't too big of a cost in terms of data returned and if it ever looks like that could be an issue, we can limit the number of queues an admin can own at a time.

> So can we dispatch at root of home page?

Do you mean change it from the way it's written in this PR? To where?

Reply:
No need to fetch every time the page is rendered. You fetch once the app is loaded. Then on successful add or delete, just update the state.